http/preview.go

```diff
@@ -80,33 +80,30 @@ func handleImagePreview(w http.ResponseWriter, r *http.Request, imgSvc ImgServic
 		return errToStatus(err), err
 	}
 
-	isFresh := checkEtag(w, r, file.ModTime.Unix(), file.Size)
-	if isFresh {
-		return http.StatusNotModified, nil
-	}
-
+	var resizedImage []byte
 	cacheKey := previewCacheKey(file.Path, file.ModTime.Unix(), previewSize)
 	cachedFile, ok, err := fileCache.Load(r.Context(), cacheKey)
 	if err != nil {
 		return errToStatus(err), err
 	}
 	if ok {
-		_, _ = w.Write(cachedFile)
-		return 0, nil
-	}
+		resizedImage = cachedFile
```

Comment:
It can be rewritten to:

```go 
resizedImage, ok, err := fileCache.Load(r.Context(), cacheKey)
if err != nil {
    return errToStatus(err), err
}
if !ok {
    resizedImage, err = createPreview(imgSvc, fileCache, file, previewSize, enableThumbnails, resizePreview)
    if err != nil { 
        return errToStatus(err), err
    }
}
```